packages/opentelemetry-scope-async-hooks/src/AsyncHooksScopeManager.ts

```diff
@@ -94,6 +116,107 @@ export class AsyncHooksScopeManager implements ScopeManager {
     return contextWrapper as any;
   }
 
+  /**
+   * By default, EventEmitter call their callback with their scope, which we do
+   * not want, instead we will bind a specific scope to all callbacks that
+   * go through it.
+   * @param target EventEmitter a instance of EventEmitter to patch
+   * @param scope the scope we want to bind
+   */
+  private _bindEventEmitter<T extends EventEmitter>(
+    target: T,
+    scope?: unknown
+  ): T {
+    const ee = (target as unknown) as PatchedEventEmitter;
+    if (ee.__ot_listeners !== undefined) return target;
+    ee.__ot_listeners = {};
+
+    // patch methods that add a listener to propagate scope
+    ADD_LISTENER_METHODS.forEach(methodName => {
+      if (ee[methodName] === undefined) return;
```

Comment:
This should also be done for the other methods, for example `removeListener`, `off`, etc.

Reply:
done